src/Symfony/Component/Form/ResolvedFormType.php

```diff
@@ -45,11 +50,18 @@ class ResolvedFormType implements ResolvedFormTypeInterface
 
     public function __construct(FormTypeInterface $innerType, array $typeExtensions = array(), ResolvedFormTypeInterface $parent = null)
     {
-        if (!preg_match('/^[a-z0-9_]*$/i', $innerType->getName())) {
+        $name = $innerType->getName();
+
+        // As of Symfony 2.8, getName() returns null by default. In this case
+        // (and as of Symfony 3.0), the FQCN of the type is used as name
+        // The deprecation error is triggered in FormRegistry
+        if (null === $name) {
+            $name = get_class($innerType);
+        } elseif (!preg_match('/^[a-z0-9_]*$/i', $name)) {
```

Comment:
@stof I just looked at this issue and you're right. Returning a FQCN from `getName()` would only work if we weakened the naming conditions in 2.3+ and handled FQCNs accordingly (e.g. when converting a type name to a form name/block name). That might break compatibility with other apps though, which rely on the current naming restrictions and e.g. cannot handle backslashes in type names.